cwatm/hydrological_modules/water_quality/erosed.py

```diff
@@ -39,15 +40,107 @@ class waterquality_erosed(object):
+        i = 0
+        for variable in overlandManningVars:
+            self.var.manOverland[i] += loadmap(variable)
+            i += 1
+        # manningsN channel
+        self.var.manNChan = loadmap('chanMan')
+        # grid slope
+        tanslope = loadmap('tanslope')
+
+        # setting slope >= 0.00001 to prevent 0 value
+        self.var.tanslope = np.maximum(tanslope, 0.00001)
+
+        # tov: time of overland ocncentration
+        tov = divideArrays(np.power(self.var.lsFactor, 0.6) * np.power(self.var.manOverland, 0.6), 18 * np.power(self.var.tanslope, 0.3))
+
+        # channel flow time of concentration
+        #tch = divideArrays(0.62 * self.var.chanLength * np.power(self.var.manNChan, 0.75), np.power(self.var.cellArea, 0.125) * np.power(self.var.chanGrad, 0.375))
+
+        #print('tch', np.nanmean(tch)
+        # tconc: time of concentration for the grid
+            # provides very high values:  axis = 0: [1930.83034355  363.97959098 1952.27786728 ... 1685.05161614 3080.56903285
+                                         #  2302.08594123]
+                                        # axis = 1: [3875.37793006 3875.34839299 3875.3501527  3875.3501527 ]
+
+        #print('tov:', np.nanmean(tov, axis=1))
+        #print('tch_min:', np.nanmin(tch), 'tch_max:', np.nanmax(tch), 'tch_mean', np.nanmean(tch))
+        #print('tconc:', np.nanmean(self.var.tconc, axis=1))
+        print('chanman min:', np.nanmin(self.var.chanMan), 'max:', np.nanmax(self.var.chanMan), 'mean:', np.nanmean(self.var.chanMan))
+        manningsFactor = loadmap('manningsN')
+        print('manningsFactor channel calibrated', manningsFactor)
+        channelmanning= loadmap('chanMan')
+        print('chanman uncalibrated .nc input min,max,mean', np.nanmin(channelmanning), np.nanmax(channelmanning),np.nanmean(channelmanning))
+        print('channel gradient min:', np.nanmin(self.var.chanGrad), 'max:', np.nanmax(self.var.chanGrad), 'mean:',
+              np.nanmean(self.var.chanGrad))
+        print('channel lenght min:', np.nanmin(self.var.chanLength), 'max:', np.nanmax(self.var.chanLength), 'mean:',
+              np.nanmean(self.var.chanLength))
+
 
     def dynamic(self):
         """
         Dynamic part of EROSED module
         """
-        print("erosed dyn")
\ No newline at end of file
+        '''
+        # Modified Universal Soil Erosion (MUSLE) for sediment yield
+        # M_(in_land)  = 11.8×〖(Q_surf*q_peak*A_grid)〗^0.56×K×C×P×LS*f_cfr
+        # 11.8 & 0.56 -> calibration parameters, call from settingsfile
+        # Q_surf surface runoff volume in mm from cwatm
+        # q_peak...peak runoff rate (m3/s); a_tc*Q_surf*Agrid/3.6*t_conc
+            #a_tc...frac. of daily rain falling in time of concentration
+            # t_conc...time of concentration for grid (model variable) hour
+        #if self.var.a
+        #self.var.runoffEnergyFactor = self.var.sum_directRunoff * 2
+        '''
+        self.waterquality_vars.dynamic()
```

Comment:
Delete from here and call in waterquality_main line 190

Reply:
Directly below the line _def dynamic(self):_ ?
line 190 does not make sense in my file.

Reply:
See below:

` def dynamic(self): 
        if dateVar['newStart'] or dateVar['newYear']:
            current_year = globals.dateVar['currDate']
            self.var.fracManagedGrassland = globals.inZero.copy()
            if 'fractionManagedGrassland' in binding:
                self.var.fracManagedGrassland = loadmap('fractionManagedGrassland')
        
        
        # Calculate gross GW recharge from 3rd soil layer for water quality
        self.var.grossGWrechargeFromSoil =  (self.var.gwRecharge + self.var.capRiseFromGW) * divideArrays(self.var.perc3toGW, self.var.perc3toGW + self.var.prefFlow)
        self.var.grossGWrechargeFromPrefFlow = (self.var.gwRecharge + self.var.capRiseFromGW) * divideArrays(self.var.prefFlow, self.var.perc3toGW + self.var.prefFlow)
        *** HERE***
        # run initial sub-modules
        # Run the Erosion and Sediment Yield (EroSed) module`

Reply:
Got it, thanks!